moveit_ros/hybrid_planning/test/test_basic_integration.cpp

```diff
@@ -0,0 +1,272 @@
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Andy Zelenak
+   Desc:   Launch hybrid planning and test basic functionality
+*/
+
+#include <gtest/gtest.h>
+#include <moveit/kinematic_constraints/utils.h>
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit/robot_state/conversions.h>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_action/rclcpp_action.hpp>
+#include <tf2_ros/buffer.h>
+
+#include <moveit_msgs/action/hybrid_planner.hpp>
+#include <moveit_msgs/msg/display_robot_state.hpp>
+#include <moveit_msgs/msg/motion_plan_response.hpp>
+
+namespace moveit_hybrid_planning
+{
+using namespace std::chrono_literals;
+
+class HybridPlanningFixture : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+  }
+
+  HybridPlanningFixture() : node_(std::make_shared<rclcpp::Node>("hybrid_planning_testing"))
+  {
+    action_successful_ = false;
+    action_aborted_ = false;
+    action_complete_ = false;
+
+    executor_.add_node(node_);
+
+    std::string hybrid_planning_action_name = "";
+    node_->declare_parameter("hybrid_planning_action_name", "");
+    if (node_->has_parameter("hybrid_planning_action_name"))
+    {
+      node_->get_parameter<std::string>("hybrid_planning_action_name", hybrid_planning_action_name);
+    }
+    else
+    {
+      RCLCPP_ERROR(node_->get_logger(), "hybrid_planning_action_name parameter was not defined");
+      std::exit(EXIT_FAILURE);
+    }
+
+    hp_action_client_ =
+        rclcpp_action::create_client<moveit_msgs::action::HybridPlanner>(node_, hybrid_planning_action_name);
+    robot_state_publisher_ = node_->create_publisher<moveit_msgs::msg::DisplayRobotState>("display_robot_state", 1);
+
+    // Add new collision object as soon as global trajectory is available.
+    global_solution_subscriber_ = node_->create_subscription<moveit_msgs::msg::MotionPlanResponse>(
+        "global_trajectory", rclcpp::SystemDefaultsQoS(),
+        [this](const moveit_msgs::msg::MotionPlanResponse::SharedPtr msg) {});
+
+    RCLCPP_INFO(node_->get_logger(), "Initialize Planning Scene Monitor");
+    tf_buffer_ = std::make_shared<tf2_ros::Buffer>(node_->get_clock());
+
+    planning_scene_monitor_ = std::make_shared<planning_scene_monitor::PlanningSceneMonitor>(
+        node_, "robot_description", tf_buffer_, "planning_scene_monitor");
+    if (!planning_scene_monitor_->getPlanningScene())
+    {
+      RCLCPP_ERROR(node_->get_logger(), "The planning scene was not retrieved!");
+      std::exit(EXIT_FAILURE);
+    }
+    else
+    {
+      planning_scene_monitor_->startStateMonitor();
+      planning_scene_monitor_->providePlanningSceneService();  // let RViz display query PlanningScene
```

Comment:
is this needed for your test in some way?

Reply:
nope, just copied from the demo